### PR TITLE
Add Migration docs

### LINKED ISSUE #38: MGW Migration Document is Missing - MGW 3.2.0
### Description:

By default, Microgateway does not require any migration process. We can build the mgw project using the Microgateway 3.2.0 toolkit and run with the Microgateway 3.2.0 runtime with the old configurations. (Ex. build 3.1.0 project with 3.2.0 toolkit and run with 3.2.0 runtime with the 3.1.0 configs)
All most all of the configurations are backwards compatible and there are only few config level changes which will be updated in the docs. (related to authentication and subscription validation).

Migration would only be required if we have any customizations written in ballerina.

But this explanation is not available in the document[1]. 

We need to get it added to the docs. Even if there are zero steps to be followed, that itself should be mentioned in the docs under migration because otherwise, the users will be in doubt. 

Therefore please add this document before releasing the product. 

[1]. https://mg.docs.wso2.com/en/latest/

### Content Positioning in Documentation:

- Link:

- Heading (& Step):

- Any other reference:

---

### PR DESCRIPTION
**Pull request by menakaj**
* Add migration from previous versions page
* Fix Configuration with API manager page

### Issues

Fixes #38

---
#### Maintainers: Check before merge
- [ ] Manual verification

### CODE REVIEW

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Migrating from Previous Versions
+
+Microgateway consists of main two distributions.
+
+- Microgateway Toolkit : Used to build the Microgateway project and generate the Microgateway executable.
+- Microgateway Runtime : Run the Microgateway executable built with the toolkit applying the requried configurations.
+
+Each Microgateway version has it's own toolkit and runtime. And the Migrogateway executable which is built from specific toolkit version is specific only to that Microgateway version.
+
+If you want to use the same Microgateway (built from a older Microgateway toolkit version) with Microgateway 3.2.0, follow the steps below.
```
Comment (Rajith90) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:10]:
```suggestion
If you want to use the same Microgateway project (built from an older Microgateway toolkit version) with Microgateway 3.2.0, follow the steps below.
```

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+1. Build the initial Microgateway Project by using the Microgateway Toolkit 3.2.0.
+2. Copy the configuration from {OLD_MGW_RUNTIME_HOME}/conf/micro-gw.conf and apply them in {MGW_3.2.0_RUNTIME_HOME}/conf/micro-gw.conf file. Refer to the [micro-gw.conf]({{base_url}}/../reference/configurations/micro-gw.conf/) and do any changes to the old configurations.
+3. If you use API Manager along with Microgateway for key validation, throttling etc, follow the [Configuration for WSO2 API Manager]({{base_url}}/../install-and-setup/configuration-for-wso2-api-manager/) document and do the relevent configuration in Microgateway 3.2.0 runtime.
+
+    !!! note
+        If you have implemented any custom ballerina extensions for Microgateway, you will have to re-write them with the latest ballerina version.
```
Comment (Rajith90) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:17]: Lets mention the new ballerina version which is compatible

Comment (menakaj) [reply en/docs/install-and-setup/migrating-from-previous-versions.md]: Fixed in https://github.com/wso2/docs-mg/pull/40/commits/b28a92a0c430d10e2d6f1abcaa9444374d0fa998

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 3]
```diff
@@ -1,62 +1,72 @@
...
 
     ``` java
         keytool -export -alias wso2carbon -file wso2carbon.crt -keystore <API-M_HOME>/repository/resources/security/wso2carbon.jks
     ```
 
-        Import the certificate to the trust store in the Microgateway Toolkit.
+    - Import the certificate to the trust store in the Microgateway Toolkit.
 
     ``` java
         keytool -import -trustcacerts -alias wso2carbon2 -file wso2carbon.crt -keystore <MGW_TOOLKIT_HOME>/lib/platform/bre/security/ballerinaTruststore.p12
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:32]:
```suggestion
    ```tab="Example"
    keytool -import -trustcacerts -alias wso2carbon2 -file wso2carbon.crt -keystore <MGW_TOOLKIT_HOME>/lib/platform/bre/security/ballerinaTruststore.p12
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 4]
```diff
@@ -1,62 +1,72 @@
...
-        !!! info
-        Refer the documentation on [Importing Certificates to the API Microgateway](https://docs.wso2.com/display/MG310/Importing+Certificates+to+the+API+Microgateway+Truststore) [Truststore](https://docs.wso2.com/display/MG310/Importing+Certificates+to+the+API+Microgateway+Truststore) for more information.
+    - Export public certificate of WSO2 API Manager.
 
-    Export public certificate of WSO2 API Manager.
     A sample command is given below.
 
-    ``` java
+    ```java
         keytool -export -alias wso2carbon -file wso2carbon.crt -keystore <API-M_HOME>/repository/resources/security/wso2carbon.jks
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:57]:
```suggestion
    ```tab="Example"
    keytool -export -alias wso2carbon -file wso2carbon.crt -keystore <API-M_HOME>/repository/resources/security/wso2carbon.jks
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 5]
```diff
@@ -1,62 +1,72 @@
...
-        1.  Add the public certificate of WSO2 API Manager to the trust store.
+    !!! info
+        Refer the documentation on [Importing Certificates to the API Microgateway]({{base_url}}/how-tos/security/importing-certificates-to-the-api-microgateway-truststore/) for more information.
 
-        !!! info
-        Refer the documentation on [Importing Certificates to the API Microgateway](https://docs.wso2.com/display/MG310/Importing+Certificates+to+the+API+Microgateway+Truststore) [Truststore](https://docs.wso2.com/display/MG310/Importing+Certificates+to+the+API+Microgateway+Truststore) for more information.
+    - Export public certificate of WSO2 API Manager.
 
-    Export public certificate of WSO2 API Manager.
     A sample command is given below.
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:54]:
```suggestion
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 6]
```diff
@@ -1,62 +1,72 @@
...
     !!! info
-        Refer the documentation on [Importing Certificates to the API Microgateway](https://docs.wso2.com/display/MG310/Importing+Certificates+to+the+API+Microgateway+Truststore) [Truststore](https://docs.wso2.com/display/MG310/Importing+Certificates+to+the+API+Microgateway+Truststore) for more information.
+        Refer the documentation on [Importing Certificates to the API Microgateway Truststore]({{base_url}}/how-tos/security/importing-certificates-to-the-api-microgateway-truststore/) for more information.
 
-    Export the public certificate of WSO2 API Manager.
+    - Export the public certificate of WSO2 API Manager.
     A sample command is given below.
 
     ``` java
         keytool -export -alias wso2carbon -file wso2carbon.crt -keystore <API-M_HOME>/repository/resources/security/wso2carbon.jks
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:26]:
```suggestion

    ```tab="Example"
    keytool -export -alias wso2carbon -file wso2carbon.crt -keystore <API-M_HOME>/repository/resources/security/wso2carbon.jks
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 7]
```diff
@@ -1,62 +1,72 @@
...
-    Export public certificate of WSO2 API Manager.
     A sample command is given below.
 
-    ``` java
+    ```java
         keytool -export -alias wso2carbon -file wso2carbon.crt -keystore <API-M_HOME>/repository/resources/security/wso2carbon.jks
     ```
 
-        Import the certificate using the command given below, to the trust store in the Microgateway runtime.
+    - Import the certificate using the command given below, to the trust store in the Microgateway runtime.
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:60]:
```suggestion
    - Import the certificate to the trust store of the Microgateway Runtime.
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 8]
```diff
@@ -1,62 +1,72 @@
...
 
-    ``` java
+    ```java
         keytool -import -trustcacerts -alias wso2carbon2 -file wso2carbon.crt -keystore <MGW_RUNTIME_HOME>/runtime/bre/security/ballerinaTruststore.p12
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:63]:
```suggestion
    ```tab="Example"
    keytool -import -trustcacerts -alias wso2carbon2 -file wso2carbon.crt -keystore <MGW_RUNTIME_HOME>/runtime/bre/security/ballerinaTruststore.p12
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 9]
```diff
@@ -1,62 +1,72 @@
...
     ```
 
-        2.  Open the `           <MGW_RUNTIME_HOME>/conf/micro-gw.conf          ` file. Add the alias (e.g. wso2carbon2) to the truststore to corresponding configurations.
-        For example, to configure the JWT authentication for API Manager JWTs, add the following JWT issuer configuration to the `           micro-gw.conf          ` .
+2.  Open the `<MGW_RUNTIME_HOME>/conf/micro-gw.conf` file. Add the alias (e.g. wso2carbon2) to the truststore to corresponding configurations.
+    
+    For example, to configure the JWT authentication for API Manager JWTs, add the following JWT issuer configuration to the `micro-gw.conf` .
+
 
     ``` java
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:71]:
```suggestion
    ```tab="Example"
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 10]
```diff
@@ -66,4 +76,54 @@ Follow the instructions below to configure the WSO2 Microgateway Toolkit and th
         validateSubscription = false
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:76]: Change the indentation inside the block as below
```
    ```tab="Example"
    [[jwtTokenConfig]]
        issuer="https://localhost:9443/oauth2/token"
        audience="http://org.wso2.apimgt/gateway"
        certificateAlias="wso2carbon2"
        validateSubscription = false
    ```
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 11]
```diff
@@ -66,4 +76,54 @@ Follow the instructions below to configure the WSO2 Microgateway Toolkit and th
...
     ```
 
-        You can now successfully invoke APIs through the Microgateway using the JWT tokens issued by WSO2 API Manager.
+You can now successfully invoke APIs through the Microgateway using the JWT tokens issued by WSO2 API Manager.
+
+## Configuring Microgateway 3.2.x Runtime with Older API Manager Versions
+
+When you use Microgateway 3.2.0 runtime with older versions of WSO2 API Manager (2.6.0, 3.0.0 or 3.1.0), please do the below configurations in the {MGW_RUNTIME_HOME}/conf/micro-gw.conf file.
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:83]:
```suggestion
When you use Microgateway 3.2.0 runtime with older versions of WSO2 API Manager (2.6.0, 3.0.0 or 3.1.0), please do the below configurations in the `<MGW_RUNTIME_HOME>/conf/micro-gw.conf` file.
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 12]
```diff
@@ -66,4 +76,54 @@ Follow the instructions below to configure the WSO2 Microgateway Toolkit and th
...
+
+#### Enable Legacy Key Manager
+
+``` java
+    [keyManager]
+    enableLegacyMode = true
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:89]:
```suggestion
```tab="Example"
[keyManager]
    enableLegacyMode = true
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 13]
```diff
@@ -66,4 +76,54 @@ Follow the instructions below to configure the WSO2 Microgateway Toolkit and th
...
+```
+
+This enables using the API Manager key validation service for opaque token validation. 
+
+#### Disable Event Hub Configuration
+
+``` java
+    [apim.eventHub]
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:97]:
```suggestion
```tab="Example"
[apim.eventHub]
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 14]
```diff
@@ -66,4 +76,54 @@ Follow the instructions below to configure the WSO2 Microgateway Toolkit and th
...
+    enable = false
+```
+
+Apim Event hub is used for retrieving API, Application and subscription created in API Manager. This is only available in API Manager 3.2.0.
+
+#### Change the Throttling Server URL
+
+``` java
+    [throttlingConfig.dataRetriever]
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:106]:
```suggestion
```tab="Example"
[throttlingConfig.dataRetriever]
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 15]
```diff
@@ -66,4 +76,54 @@ Follow the instructions below to configure the WSO2 Microgateway Toolkit and th
...
+    serverUrl = "https://{API-M_HOST}:{API-M_PORT}/throttle/data/v1"
+```
+
+#### Change the Token Revocation Endpoint URL
+
+``` java
+    [tokenRevocationConfig.persistent]
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:113]:
```suggestion
```tab="Example"
[tokenRevocationConfig.persistent]
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 16]
```diff
@@ -66,4 +76,54 @@ Follow the instructions below to configure the WSO2 Microgateway Toolkit and th
...
+    endpointURL = "https://{API-M_HOST}:{API-M_PORT}/throttle/data/v1"
+```
+
+#### Set the Analytics Stream Version
+
+``` java
+    [analytics]
```
Comment (praminda) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:120]:
```suggestion
```tab="Example"
[analytics]
```

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 17]
```diff
@@ -0,0 +1,17 @@
+# Migrating from Previous Versions
+
+Microgateway consists of main two distributions.
+
+- Microgateway Toolkit : Used to build the Microgateway project and generate the Microgateway executable.
```
Comment (praminda) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:5]:
```suggestion
- Microgateway Toolkit : Used to build the Microgateway project and generate the Microgateway runtime artifacts.
```

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+- Microgateway Runtime : Run the Microgateway executable built with the toolkit applying the requried configurations.
+
+Each Microgateway version has it's own toolkit and runtime. And the Migrogateway executable which is built from specific toolkit version is specific only to that Microgateway version.
+
+If you want to use the same Microgateway project (built from an older Microgateway toolkit version) with Microgateway 3.2.0, follow the steps below.
+
+1. Build the initial Microgateway Project by using the Microgateway Toolkit 3.2.0.
+2. Copy the configuration from {OLD_MGW_RUNTIME_HOME}/conf/micro-gw.conf and apply them in {MGW_3.2.0_RUNTIME_HOME}/conf/micro-gw.conf file. Refer to the [micro-gw.conf]({{base_url}}/../reference/configurations/micro-gw.conf/) and do any changes to the old configurations.
+3. If you use API Manager along with Microgateway for key validation, throttling etc, follow the [Configuration for WSO2 API Manager]({{base_url}}/../install-and-setup/configuration-for-wso2-api-manager/) document and do the relevent configuration in Microgateway 3.2.0 runtime.
```
Comment (praminda) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:14]:
```suggestion
2. Copy the configuration from `<OLD_MGW_RUNTIME_HOME>/conf/micro-gw.conf` and apply them in `<MGW_3.2.0_RUNTIME_HOME>/conf/micro-gw.conf` file. Refer to the [micro-gw.conf](/reference/configurations/micro-gw.conf/) and do any changes to the old configurations.
3. If you use API Manager along with Microgateway for key validation, throttling etc, follow the [Configuration for WSO2 API Manager](/install-and-setup/configuration-for-wso2-api-manager/) document and do the relevant configuration in Microgateway 3.2.0 runtime.
```

**Review by praminda — CHANGES_REQUESTED** [review]

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md [hunk 19]
```diff
@@ -1,69 +1,129 @@
...
+```tab="Example"
+    [throttlingConfig.dataRetriever]
+    serverUrl = "https://{API-M_HOST}:{API-M_PORT}/throttle/data/v1"
+```
+
+#### Change the Token Revocation Endpoint URL
+
+```tab="Example"
+    [tokenRevocationConfig.persistent]
+    endpointURL = "https://{API-M_HOST}:{API-M_PORT}/throttle/data/v1"
```
Comment (AmaliMatharaarachchi) [inline en/docs/install-and-setup/configuration-for-wso2-api-manager.md:114]: Is this example or format of the config? {API-M_HOST}:{API-M_PORT} are not known properties

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+Each Microgateway version has it's own toolkit and runtime. And the Migrogateway executable which is built from specific toolkit version is specific only to that Microgateway version.
+
+If you want to use the same Microgateway project (built from an older Microgateway toolkit version) with Microgateway 3.2.0, follow the steps below.
+
+1. Build the initial Microgateway Project by using the Microgateway Toolkit 3.2.0.
+2. Copy the configuration from `<OLD_MGW_RUNTIME_HOME>/conf/micro-gw.conf` and apply them in `<MGW_3.2.0_RUNTIME_HOME>/conf/micro-gw.conf` file. Refer to the [micro-gw.conf](/reference/configurations/micro-gw.conf/) and do any changes to the old configurations.
+3. If you use API Manager along with Microgateway for key validation, throttling etc, follow the [Configuration for WSO2 API Manager](/install-and-setup/configuration-for-wso2-api-manager/) document and do the relevant configuration in Microgateway 3.2.0 runtime.
+
+    !!! note
+        If you have implemented any custom ballerina extensions for Microgateway previously, you will have to re-write them with the ballerina version 1.2.7.
```
Comment (AmaliMatharaarachchi) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:17]: shall we add the link to ballerina API docs?

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 21]
```diff
@@ -0,0 +1,17 @@
+# Migrating from Previous Versions
+
+Microgateway consists of main two distributions.
+
+- Microgateway Toolkit : Used to build the Microgateway project and generate the Microgateway executable.
+- Microgateway Runtime : Run the Microgateway executable built with the toolkit applying the requried configurations.
```
Comment (AmaliMatharaarachchi) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:6]: * requried
```suggestion
- Microgateway Runtime : Run the Microgateway executable built with the toolkit applying the required configurations.
```

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+
+Each Microgateway version has it's own toolkit and runtime. And the Migrogateway executable which is built from specific toolkit version is specific only to that Microgateway version.
```
Comment (AmaliMatharaarachchi) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:8]: *it's *specific
```suggestion
Each Microgateway version has its own toolkit and runtime. And the Migrogateway executable which is built from a specific toolkit version is specific only to that Microgateway version.
```

File: en/docs/install-and-setup/migrating-from-previous-versions.md [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+
+If you want to use the same Microgateway project (built from an older Microgateway toolkit version) with Microgateway 3.2.0, follow the steps below.
+
+1. Build the initial Microgateway Project by using the Microgateway Toolkit 3.2.0.
+2. Copy the configuration from `<OLD_MGW_RUNTIME_HOME>/conf/micro-gw.conf` and apply them in `<MGW_3.2.0_RUNTIME_HOME>/conf/micro-gw.conf` file. Refer to the [micro-gw.conf](/reference/configurations/micro-gw.conf/) and do any changes to the old configurations.
+3. If you use API Manager along with Microgateway for key validation, throttling etc, follow the [Configuration for WSO2 API Manager](/install-and-setup/configuration-for-wso2-api-manager/) document and do the relevant configuration in Microgateway 3.2.0 runtime.
```
Comment (AmaliMatharaarachchi) [inline en/docs/install-and-setup/migrating-from-previous-versions.md:14]:
```suggestion
3. If you use API Manager along with Microgateway for key validation, throttling, etc, follow the [Configuration for WSO2 API Manager](/install-and-setup/configuration-for-wso2-api-manager/) document and do the relevant configuration in Microgateway 3.2.0 runtime.
```

File: en/docs/install-and-setup/configuration-for-wso2-api-manager.md (hunk 19 above)
Comment (menakaj) [reply en/docs/install-and-setup/configuration-for-wso2-api-manager.md]: Fixed in https://github.com/wso2/docs-mg/pull/40/commits/ce01f07681de7fbc80cd3d29cb95bf84c254aa45

File: en/docs/install-and-setup/migrating-from-previous-versions.md (hunk 20 above)
Comment (menakaj) [reply en/docs/install-and-setup/migrating-from-previous-versions.md]: Added in https://github.com/wso2/docs-mg/pull/40/commits/438a39caee3bb53961579af42bdbf584b136914b

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Review by praminda — APPROVED** [review]

**Pull request merged**